pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -262,11 +262,14 @@ func calculateIPOffset(base *big.Int, ip net.IP) int {
 // RangeSize returns the size of a range in valid addresses.
 func RangeSize(subnet *net.IPNet) int64 {
 	ones, bits := subnet.Mask.Size()
-	if bits == 32 && (bits-ones) >= 31 || bits == 128 && (bits-ones) >= 63 {
+	if bits == 32 && (bits-ones) >= 31 || bits == 128 && (bits-ones) >= 121 {
 		return 0
 	}
-	max := int64(1) << uint(bits-ones)
-	return max
+	if bits == 128 && (bits-ones) >= 16 {
```

Comment:
I do not understand this logic?  It needs at least some comments.  Where is "16" coming from?  When will this be fixed, who is responsible, etc.

Reply:
Thanks for the review @thockin The 16 was used because it is similar to changes in cidr_set
With a max of 16, the bitmap can grow to a maximum of 2**16, or 64k bits, which uses 1k uint64's. If it is allowed to get bigger, say a /96, that works to 2**32, 67 million uint64's.
I will add comments for this.

As for the future fix, would this code be impacted by (#57130) ?

Also, the reason for this intermediate fix is to get rid of the /66 restriction for IPv6 beta. The /66 restriction is not very practical to implement.

Reply:
I figured out the 16 eventually but it needed to be commented.  Re-reviewing now.